glean-core/tests/ping.rs

```diff
@@ -101,3 +102,89 @@ fn empty_pings_with_flag_are_sent() {
     assert_eq!(false, ping2.submit(&glean, None).unwrap());
     assert_eq!(1, get_queued_pings(glean.get_data_path()).unwrap().len());
 }
+
+#[test]
+fn test_pings_submitted_metric() {
+    let (mut glean, _temp) = new_glean(None);
+
+    let pings_submitted = LabeledMetric::new(
+        CounterMetric::new(CommonMetricData {
+            name: "pings_submitted".into(),
+            category: "glean.validation".into(),
+            send_in_pings: vec!["metrics".into(), "baseline".into()],
+            lifetime: Lifetime::Ping,
+            disabled: false,
+            dynamic_label: None,
+        }),
+        None,
+    );
+
+    let metrics_ping = PingType::new("metrics", true, false, vec![]);
+    glean.register_ping_type(&metrics_ping);
+
+    let baseline_ping = PingType::new("baseline", true, false, vec![]);
+    glean.register_ping_type(&baseline_ping);
+
+    // We need to store a metric as an empty ping is not stored.
+    let counter = CounterMetric::new(CommonMetricData {
+        name: "counter".into(),
+        category: "local".into(),
+        send_in_pings: vec!["metrics".into()],
+        ..Default::default()
+    });
+    counter.add(&glean, 1);
+
+    assert!(metrics_ping.submit(&glean, None).unwrap());
+
+    assert_eq!(
+        Some(1),
+        pings_submitted
+            .get("metrics")
+            .test_get_value(&glean, "metrics")
+    );
+    assert_eq!(
+        None,
+        pings_submitted
+            .get("baseline")
+            .test_get_value(&glean, "metrics")
+    );
+    assert_eq!(
+        Some(1),
+        pings_submitted
+            .get("metrics")
+            .test_get_value(&glean, "baseline")
+    );
+    assert_eq!(
+        None,
+        pings_submitted
+            .get("baseline")
+            .test_get_value(&glean, "baseline")
+    );
+
+    assert!(baseline_ping.submit(&glean, None).unwrap());
```

Comment:
Can we trigger 2 baseline pings here? Likely to happen out in the wild and then we can test that we indeed get a `2` further below.

Reply:
Actually maybe a comment here about which values we expect to reset would be helpful.